### ISSUE TITLE
Rule Request: Place access modifiers on extension methods and not on the extension itself

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2356]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

I got the inspiration from this discussion: https://github.com/apple/swift/pull/18623#issuecomment-412718344

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
public extension X {
  func foo()
}
```

Would **not** trigger:

```swift
extension X {
  public func foo()
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

I guess it could be configurable to enforce `public extension X` instead, but I don't have the need personally as I think it's better to explicitly have the access modifier on the extension methods.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

I think it should be enabled by default as I don't think it would have any false-positives and it seems generally useful to be consistent about it.

**Comment by marcelofabri** [comment]
There's already a rule for this: https://github.com/realm/SwiftLint/blob/master/Rules.md#no-extension-access-modifier
